controllers/storageconsumer/storageconsumer_controller.go

```diff
@@ -658,3 +683,26 @@ func (r *StorageConsumerReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		Owns(&rookCephv1.CephClient{}).
 		Complete(r)
 }
+
+func (c *cephClientFormat) generateHashForCephClient(storageConsumerName, claimID, cephUserType string) string {
+	c.StorageConsumerName = storageConsumerName
+	c.ClaimID = claimID
+	c.CephUserType = cephUserType
+
+	cephClient, err := json.Marshal(c)
+	if err != nil {
+		klog.Errorf("failed to marshal ceph client name for consumer %s. %v", storageConsumerName, err)
+		panic("failed to marshal")
+	}
+	name := md5.Sum([]byte(cephClient))
+	return hex.EncodeToString(name[:16])
+}
+
+func setAnnotations(storageConsumerName, storageClaim, cephUserType string) map[string]string {
+	annotations := map[string]string{
+		storageConsumerAnnotation:     storageConsumerName,
+		storageClaimAnnotation:        storageClaim,
+		storageCephUserTypeAnnotation: cephUserType,
+	}
+	return annotations
```

Comment:
Annotations are usually considered both an input/output, you cannot just override them completely - which is what this util function is doing.